### PR TITLE
[SMALLFIX] Shade protobuf in a single module

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -26,6 +26,7 @@
     <module>common</module>
     <module>client</module>
     <module>server</module>
+    <module>protobuf</module>
```
Comment (jsimsa) [inline core/pom.xml]: sort alphabetically?

File: core/protobuf/pom.xml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            <configuration>
+              <artifactSet>
+                <includes>
+                  <include>com.google.protobuf:protobuf-java</include>
+                </includes>
+              </artifactSet>
+              <relocations>
+                <relocation>
+                  <pattern>com.google.protobuf</pattern>
+                  <shadedPattern>alluxio.shaded.protobuf</shadedPattern>
```
Comment (jsimsa) [inline core/protobuf/pom.xml]: I would prefer to use "alluxio.shaded.com.google.protobuf" as the shaded pattern

File: core/server/pom.xml [hunk 3]
```diff
@@ -32,14 +32,10 @@
...
     <dependency>
-      <groupId>com.google.protobuf</groupId>
-      <artifactId>protobuf-java</artifactId>
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-core-protobuf</artifactId>
+      <version>${project.version}</version>
     </dependency>
-    <!-- alluxio-core-common should come before alluxio-core-client as alluxio-core-client shade some dependencies
-    such as protobuf. We want to use the unshaded ones in alluxio-core-server.
-    TODO(peis, jiri): Do not depend on alluxio-core-client in alluxio-core-server to avoid depending on
```
Comment (jsimsa) [inline core/server/pom.xml]: keep the TODO

File: core/pom.xml (hunk 1 above)
Comment (apc999) [inline core/pom.xml]: done

File: core/protobuf/pom.xml (hunk 2 above)
Comment (apc999) [inline core/protobuf/pom.xml]: done

File: core/server/pom.xml (hunk 3 above)
Comment (apc999) [inline core/server/pom.xml]: done

**Comment by apc999** [comment]
@jsimsa PTAL. After successfully building the project, my Intellij still unable to parse the symbols. I will investigate this issue separately.

File: integration/mesos/pom.xml [hunk 4]
```diff
@@ -39,7 +39,7 @@
     </dependency>
     <dependency>
       <groupId>com.google.protobuf</groupId>
-      <artifactId>protobuf-java</artifactId>
+       <artifactId>protobuf-java</artifactId>
```
Comment (jsimsa) [inline integration/mesos/pom.xml]: nit: undo

Comment (jsimsa) [inline integration/mesos/pom.xml]: also, if we build alluxio with -Pmesos, we end up including this dependency in the assembly uber jar ... is that okay?

**Pull request merged**